### PR TITLE
Add data export model

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR adds the new `DataExports` model which will be used to track user exports.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. In the rails console `rails c`, run `bin/rails db:migrate` to migrate the new `DataExports` table
2. Assign a user to the variable `user`
3. Set a datetime `date = Time.now + 3.days`
4. Test adding a new `DataExport` in the console with the command `DataExport.new(user: user, name: 'test name', export_type: 'sample', status: 'processing', expires_at: date, email_notification: true, export_parameters: {ids: [1,2,3]}).save`
5. Check the export exists under `user.data_exports`
6. Test that you cannot add a new `DataExport` if the following fields are empty/missing: `user`, `export_type`, `status`.
7. Test that you can add a new `DataExport` if the following fields are empty/missing: `name`, `email_notification`, `export_parameters`, `expires_at`.
8. Ensure tests run locally

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: test/fixtures/data_exports.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+data_export_one:
+  name: Data Export 1
+  export_type: sample
+  status: processing
+  export_parameters: {sample_ids: [<%= ActiveRecord::FixtureSet.identify(:sample1) %>, <%= ActiveRecord::FixtureSet.identify(:sample) %>]}
```
Comment (ericenns) [inline test/fixtures/data_exports.yml]: **nitpick (blocking):** Can you update `sample_ids` to just be `ids` as that is what you mention in your PR description.

**Review by ericenns — CHANGES_REQUESTED** [review]
This is looking good, just a few minor comments.

File: test/fixtures/data_exports.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply test/fixtures/data_exports.yml]: Thanks for catching that, fixed in 479f4ba8e2d8629ef98f5af57bd8ff0bc70fe5b8

[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply test/fixtures/data_exports.yml]: Fixed in 479f4ba8e2d8629ef98f5af57bd8ff0bc70fe5b8

Comment (ChrisHuynh333) [reply test/fixtures/data_exports.yml]: Fixed in 479f4ba8e2d8629ef98f5af57bd8ff0bc70fe5b8

File: app/services/data_exports/create_service.rb
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply app/services/data_exports/create_service.rb:1]: Yes it was, had just caught it before you commented. d85ce829b0a1d92641cdf2499a8d64295d8f662d

File: app/models/data_export.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+# entity class for DataExport
+class DataExport < ApplicationRecord
+  has_logidze
+  acts_as_paranoid
+
+  belongs_to :user
+
+  has_one_attached :file
```
Comment (ericenns) [inline app/models/data_export.rb:10]: **nitpick (blocking):** Could you add validation for `export_type` and `status` to only allow the types and status that should be supported.

**Review by ericenns — CHANGES_REQUESTED** [review]
Just a couple minor comments on validation

File: app/models/data_export.rb (hunk 2 above)
Comment (ChrisHuynh333) [reply app/models/data_export.rb:10]: Added in b84d404d6c3b808fe8db407a866810630178247e

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**